### PR TITLE
fix: Add context to GitLab and BitBucket commit status functions

### LINKED ISSUE #7113: Include context in GitLab and BitBucket commit status function return values
The function that creates a "standard" commit status record from a GitLab one in jx doesn't include the "context" (in the case of GitLab and the two BitBuckets, the field is `name`, but it's the same as `context` in GitHub, etc). We should include that.

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

The only direct reason for this is the Lighthouse BDD tests, but it's still something we should have in there anyway.

#### Which issue this PR fixes

fixes #7113

### CODE REVIEW

**Comment by abayer** [comment]
/hold

Holding until I'm sure I actually have all the GitLab provider changes needed for BDD.

File: pkg/gits/gitlab.go [hunk 1]
```diff
@@ -292,9 +292,12 @@ func (g *GitlabProvider) UpdatePullRequest(data *GitPullRequestArguments, number
 	base := data.Base
 
 	o := &gitlab.UpdateMergeRequestOptions{
-		Title:        &title,
-		Description:  &body,
-		TargetBranch: &base,
```
Comment (mrageh) [inline pkg/gits/gitlab.go:297]: Why was the `TargetBranch` removed?

**Pull request merged**